lib/services/services_linux.c

```diff
@@ -624,6 +668,8 @@ async_action_complete(mainloop_child_t *p, pid_t pid, int core, int signo,
     if (signo == 0) {
         crm_debug("%s[%d] exited with status %d", op->id, op->pid, exitcode);
         services__set_result(op, exitcode, PCMK_EXEC_DONE, NULL);
+        log_op_output(op);
+        parse_exit_reason_from_stderr(op);
```

Comment:
could make it return the reason and use above

Reply:
I wanted to avoid the current function's strdup(). We could have an alternative function that skips the strdup(), or (as you suggested) add flags for what needs duplicating and/or freeing, but I didn't want to get that complicated to start.

Reply:
Yep - for now the performance-optimized approach.
Just harder to remember. When introducing possibility to ```take-over ownership/copy and own/just use pointer``` we'd have to go over all of them anyway. As the interface would change the compiler would anyway help/force us ;-)